src/Randomizer.SMZ3.Tracking/Services/MicrophoneService.cs

```diff
@@ -87,36 +87,36 @@ public void Dispose()
         GC.SuppressFinalize(this);
     }
 
-    public ICollection<string> GetDeviceNames()
+    public Dictionary<string, string> GetDeviceDetails()
     {
-        var toReturn = new List<string>();
+        var toReturn = new Dictionary<string, string>();
         var enumerator = new MMDeviceEnumerator();
         foreach (var wasapi in enumerator.EnumerateAudioEndPoints(DataFlow.Capture, DeviceState.Active))
         {
-            toReturn.Add(wasapi.DeviceFriendlyName);
+            toReturn[wasapi.ID] = wasapi.DeviceFriendlyName;
         }
 
         return toReturn;
     }
 
-    protected virtual MMDevice? GetInputDeviceByName(string? deviceName)
+    protected virtual MMDevice? GetInputDeviceById(string? id)
     {
-        if (string.IsNullOrEmpty(deviceName) || "Default" == deviceName)
+        if (string.IsNullOrEmpty(id) || "Default" == id)
         {
             return WasapiCapture.GetDefaultCaptureDevice();
         }
 
         var enumerator = new MMDeviceEnumerator();
         return enumerator
             .EnumerateAudioEndPoints(DataFlow.Capture, DeviceState.Active)
-            .FirstOrDefault(wasapi => wasapi.DeviceFriendlyName == deviceName);
+            .FirstOrDefault(wasapi => wasapi.ID == id);
```

Comment:
There should still be a fallback to the default here in case the device gets disconnected/removed between setting it in Options and actually starting Tracker. Maybe with an error, logged and ideally shown to the user if possible.

Reply:
I already covered cases of the particular audio device not being found and no audio devices being found in GetInputDevice on lines 119 and 123. Though, I think a device can be returned but its status is marked as not active anymore, in the case of being disabled. I suppose I can try to account for that scenario as well and add logging around all this.

I'll see what I can do about displaying something to the user.